### ISSUE TITLE
When I try to [make http request and read body in workflow] I get [context cancel error]

### ISSUE THREAD

**Issue by zhao85** [cortezaproject/corteza#1830]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version of Corteza

2023.9.0

### Current Behavior

when I create a workflow to test the http request function, and I set the timeout to 1hour, but the workflow always return immediately, and can not get all lines of response body.

![image](https://github.com/cortezaproject/corteza/assets/11971297/14e46fa0-0054-4252-a999-6f06eaa0acba)

### Expected Behavior

the workflow terminate normally.

### Steps To Reproduce

I have digged into the source code , and found the reason of this bug, but I'm not sure how to fix it without memory leaking.
```

func (h httpRequestHandler) send(ctx context.Context, args *httpRequestSendArgs) (r *httpRequestSendResults, err error) {
	var (
		req *http.Request
		rsp *http.Response
	)

	r = &httpRequestSendResults{}

	req, err = h.makeRequest(ctx, args)
	if err != nil {
		return nil, err
	}

	if args.Timeout > 0 {
		var tfn context.CancelFunc
		ctx, tfn = context.WithTimeout(ctx, args.Timeout)
		req = req.WithContext(ctx)
               // ************ THE BUG is here : this defer shall cause the context to cancel before read complete
		defer tfn()
	}

	rsp, err = http.DefaultClient.Do(req)
	if err != nil {
		return
	}

	r.StatusCode = int64(rsp.StatusCode)
	r.Status = rsp.Status
	r.Headers = rsp.Header
	r.ContentLength = rsp.ContentLength
	r.ContentType = rsp.Header.Get("Content-Type")
	r.Body = rsp.Body

	return
}
```

### Environment and versions

```markdown
Ubuntu 22.04 in docker
```

**Comment by zhao85** [comment]
try to fix it as #1833

**Comment by jackmaninov** [comment]
I am seeing this issue as well, it seems the PR has not been accepted yet.
